codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/SymbolVisitor.java

```diff
@@ -0,0 +1,518 @@
+                .definitionFile("./api_client.go")
+                .build();
+    }
+
+    @Override
+    public Symbol stringShape(StringShape shape) {
+        if (shape.hasTrait(EnumTrait.class)) {
+            String name = getDefaultShapeName(shape);
+            return symbolBuilderFor(shape, name, typesPackageName)
+                    .definitionFile("./types/enums.go")
+                    .build();
+        }
+
+        return symbolBuilderFor(shape, "string")
+                .putProperty(SymbolUtils.GO_UNIVERSE_TYPE, true)
+                .build();
+    }
+
+    @Override
+    public Symbol structureShape(StructureShape shape) {
+        String name = getDefaultShapeName(shape);
+        String typeName = OperationIndex.of(model).isOutputStructure(shape) ? "Result" : name;
+//        if (shape.getId().getNamespace().equals(CodegenUtils.getSyntheticTypeNamespace())) {
+//            Optional<String> boundOperationName = getNameOfBoundOperation(shape);
+//            if (boundOperationName.isPresent()) {
+//                return symbolBuilderFor(shape, name, rootModuleName)
+//                        .definitionFile("./api_op_" + boundOperationName.get() + ".go")
+//                        .build();
+//            }
+//        }
+        Symbol.Builder builder = symbolBuilderFor(shape, name, typesPackageName);
+                                         //.putProperty("dafny", symbolBuilderFor(shape, String.format("%s%s", , typeName))
+                                         //                           .build();
+        if (shape.hasTrait(ErrorTrait.ID)) {
+            builder.definitionFile("./types/errors.go");
+        } else {
+            builder.definitionFile("./types/types.go");
+        }
+
+        return builder.build();
+    }
+
+    private Optional<String> getNameOfBoundOperation(StructureShape shape) {
+        NeighborProvider provider = NeighborProviderIndex.of(model).getReverseProvider();
+        for (Relationship relationship : provider.getNeighbors(shape)) {
+            RelationshipType relationshipType = relationship.getRelationshipType();
+            if (relationshipType == RelationshipType.INPUT || relationshipType == RelationshipType.OUTPUT) {
+                return Optional.of(getDefaultShapeName(relationship.getNeighborShape().get()));
+            }
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Symbol unionShape(UnionShape shape) {
+        String name = getDefaultShapeName(shape);
+        return symbolBuilderFor(shape, name, typesPackageName)
+                .definitionFile("./types/types.go")
+                .build();
+    }
+
+    @Override
+    public Symbol memberShape(MemberShape member) {
+        Shape targetShape = model.expectShape(member.getTarget());
+        return toSymbol(targetShape)
+                .toBuilder()
+                //.putProperty(SymbolUtils.POINTABLE, pointableIndex.isPointable(member))
```

Comment:
Did we comment this out or did Smithy-Go's authors?
If it was us, why?

Also, are we avoiding pointers?
That seems to be a theme of these commented out lines...
but I am not sure...

Reply:
Same as above, added this back.